Src/Buddy.UniversalWindows/MessagingMqttClient.cs

```
@@ -0,0 +1,211 @@
+                return;
+
+            if (client.IsConnected)
+            {
+                client.Disconnect();
+                client.MqttMsgPublished -= Client_MqttMsgPublished;
+                client.MqttMsgPublishReceived -= Client_MqttMsgReceived;
+                client.ConnectionClosed -= Client_ConnectionClosed;
+                client = null;
+                serverInitiated = false;
+            }
+
+            disconnected = true;
+        }
+
+        public void Init(string token, QualityOfService qos)
+        {
+            if (client == null)
+            {
+                if (!string.IsNullOrEmpty(token))
+                {
+                    deviceToken = token;
+                }
+                else
+                {
+                    throw new ArgumentException(NO_DEVICE_TOKEN_ERROR);
+                }
+
+                qualityOfService = qos;
+
+                string address = SERVER_ADDRESS;
+                client = new MqttClient(address, SERVER_PORT, true, MqttSslProtocols.TLSv1_2);
+                client.MqttMsgPublished += Client_MqttMsgPublished;
+                client.MqttMsgPublishReceived += Client_MqttMsgReceived;
+            }
+
+            serverInitiated = true;
+
+            if (!client.IsConnected)
+            {
+                ConnectToServer();
+            }
+        }
+
+        private void ConnectToServer()
+        {
+            if (!client.IsConnected)
+            {
+                try
+                {
+                    if (!string.IsNullOrEmpty(deviceToken))
+                    {
+                        client.Connect(deviceToken);
+                    }
+                    else
+                    {
+                        throw new ArgumentException(NO_DEVICE_TOKEN_ERROR);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    throw new Exception("Error connecting to server : " + ex.Message);
```

If you're going to catch and throw, should throw a custom exception,  not just add a message.